tree_partitioning/classes/_pp_utils.py

```diff
@@ -82,18 +78,47 @@ def _netdict_from_pp_net(net, merge_lines):
 
     # Create df of the buses
     df_bus = net.res_bus
-    df_bus.loc[net.gen.bus, "p_gen"] = -net.res_gen["p_mw"].values
-    df_bus.loc[net.gen.bus, "min_p_mw"] = -net.gen["min_p_mw"].values
-    df_bus.loc[net.gen.bus, "max_p_mw"] = -net.gen["max_p_mw"].values
+    df_bus["p_mw"] = df_bus["p_mw"]  # positive is net consumption
+    df_bus.loc[net.gen.bus, "p_gen"] = net.res_gen["p_mw"].values
+    df_bus.loc[net.gen.bus, "min_p_mw"] = net.gen["min_p_mw"].values
+    df_bus.loc[net.gen.bus, "max_p_mw"] = net.gen["max_p_mw"].values
     df_bus.loc[net.load.bus, "p_load"] = net.res_load["p_mw"].values
```

Comment:
By definition, every distinct power injection component is now positive.

The total power generation `df_bus['p_mw']` is total power load - total power generation.